core/domain/html_validation_service.py

```diff
@@ -703,3 +703,31 @@ def is_parsable_as_xml(xml_string):
         return True
     except xml.etree.ElementTree.ParseError:
         return False
+
+
+def convert_svg_diagram_tags_to_image_tags(html_string):
+    """Renames all the oppia-noninteractive-svgdiagram on the
+       server to oppia-noninteractive-image and changes
+       corresponding attributes
+
+    Args:
+        html_string: str. The HTML string to check.
+
+    Returns:
+        str. The updated html string.
+    """
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
+
+    for image in soup.findAll(name='oppia-noninteractive-svgdiagram'):
+        # The 'filepath-with-value' and 'alt-with-value'
+        # attribute values should be enclosed in
+        # double quotes(&amp;quot;).
+        attr_value = escape_html(image['svg_filename-with-value'])
+        alt_value = escape_html(image['alt-with-value'])
+        del image['svg_filename-with-value']
+        image['filepath-with-value'] = attr_value
+        image['caption-with-value'] = ''
```

Comment:
Does this need to be assigned if the caption is empty?

Reply:
In all the noninteractive-images there is 'caption-with-value' attribute which can be empty(it is an optional field). But it should be there in order to initialize the image. We do not have any such caption/title attribute for svg-diagrams which could be given to the diagrams as captions and thus all the SVG-diagrams can be converted into non-interactive images with the empty field value of the 'caption-with-value' attribute.

Reply:
Thanks for the additional context, but regarding your bit about "Bit it should be there in order to initialize the image"--why? That's the main part that I'm interested in understanding. If it's an optional field, why initialize it?

Reply:
When we add an image (noninteractive-image) without any caption, still the tag is present as an empty string:
![Screenshot from 2021-05-01 23-07-28](https://user-images.githubusercontent.com/68547101/116804509-ef883280-ab3c-11eb-95a3-664f758872f2.png). Thus the default value of the attribute should be an empty string.
So, after discussion with Kevin, we are now initializing it as an empty string to avoid any errors in the RTE while editing the SVG diagrams as images.

Reply:
Ah, this additional context is really helpful. In that case, the current solution LGTM. Thanks!